deployment-aws/puppet/mevapi/manifests/cloudwatch_agent.pp

```diff
@@ -0,0 +1,38 @@
+class mevapi::cloudwatch_agent () {
+
+  if $mevapi::platform == 'aws' {
+
+    $cloudwatch_agent_dir = '/opt/aws/amazon-cloudwatch-agent'
+    $cloudwatch_agent_deb = '/tmp/amazon-cloudwatch-agent.deb'
+
+    file { $cloudwatch_agent_deb:
+        ensure => present,
+        source => 'https://s3.amazonaws.com/amazoncloudwatch-agent/ubuntu/amd64/latest/amazon-cloudwatch-agent.deb',
+    }
+
+    package { 'amazon-cloudwatch-agent':
+        provider => dpkg,
+        source   => $cloudwatch_agent_deb
+    }
+    ->
+    file { "${cloudwatch_agent_dir}/etc/amazon-cloudwatch-agent.json":
+        ensure  => file,
+        content => epp('mevapi/cloudwatch_config.json.epp'),
+    }
+    ~>
+    exec { 'cloudwatch_agent':
+        # Note that the config file (amazon-cloudwatch-agent.json) controls which log files
+        # and machine metrics are monitored. Metrics are kept for 15 days on a rolling basis
+        # https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#Metric
```

Comment:
Nit: I think this would be more useful in `cloudwatch_config.json.epp` because that's where one would look for the retention period value.

Reply:
are comments valid in json?

Reply:
yeah, that's probably a better spot